scenes/lessons/7th_class/a-b_squared/a-b_squared.py

```diff
@@ -0,0 +1,673 @@
+        self.last_exercise() # (2-y)^2 = ?
+
+    def recap_and_get_stuck(self):
+        '''
+            Writes 51^2 = (50+1)^2 = 50^2+2•50•1+1^2=2601 as we learned in previous lessons
+            Uses same method to calculate 49^2, gets (40+9)^2 = 40^2+2•40•9+9^2=2401
+            Tries to calculate 49^2 by another approach - (50-1)^2 and gets stuck
+        '''
+        # 51^2 = (50+1)^2 = 50^2 + 2•50•1 + 1^2
+        fifty_one_squared = Tex(
+            '$51^2$', ' $=$ ', '$(50+1)^2$', ' $=$ ', '$50^2 + 2$$\cdot$$50$$\cdot$$1 + 1^2$', ' $=$ ', '$2601$',
+            font_size=60
+        )
+        fifty_one_squared.shift(UP).to_edge(LEFT)
+        self.play(Write(fifty_one_squared[0]))
+        self.wait(0.25)
+        self.play(Write(fifty_one_squared[1:3]))
+        self.wait(0.25)
+        self.play(Write(fifty_one_squared[3:]))
+        self.wait()
+
+        # 49^2 = (40+9)^2 = 40^2 + 2•40•9 + 9^2
+        forty_nine_squared_1 = Tex(
+            '$49^2$', ' $=$ ', '$(40+9)^2$',
+            ' $=$ ', '$40^2$', '$+ 2$$\cdot$$40$$\cdot$$9$', '$+ 9^2$',
+            ' $=$ ', '$2401$',
+            font_size=60
+        ) # 49^2 = (40+9)^2 = 40^2 +2•40•9 + 9^2 = 2401
+        forty_nine_squared_1.next_to(fifty_one_squared, DOWN, buff=0.75, aligned_edge=LEFT)
+        calculations = VGroup(*Tex('$1600 + 720 + 81$', font_size=60)[0])
+        calculations.next_to(forty_nine_squared_1[4:7], DOWN, buff=0.5)
+
+        self.play(Write(forty_nine_squared_1[0]))
+        self.wait(0.25)
+        self.play(Write(forty_nine_squared_1[1:3]))
+        self.wait(0.25)
+        for i in [3, 4, 5, 6]:
+            self.play(Write(forty_nine_squared_1[i]))
+            self.wait(0.25)
+        self.play(Write(calculations[0:4]))
+        self.wait(0.5)
+        self.play(Write(calculations[4:8]))
+        self.wait(0.5)
+        self.play(Write(calculations[8:]))
+        self.wait(0.5)
+
+        self.play(Transform(calculations, Tex('$2401$', font_size=60).move_to(calculations)))
+        self.wait()
+        self.play(
+            Write(forty_nine_squared_1[-2]),
+            ReplacementTransform(calculations, forty_nine_squared_1[-1:])
+        )
+        self.wait()
+
+        # try another approach and get stuck - 49^2=(50-1)^2
+        self.forty_nine_squared.next_to(forty_nine_squared_1, DOWN, buff=0.75, aligned_edge=LEFT)
+        self.play(Write(self.forty_nine_squared[0:2]))
+        self.wait(0.5)
+        self.play(Write(self.forty_nine_squared[2:]))
+        self.wait()
+
+        self.play(Circumscribe(self.forty_nine_squared[4:7], fade_out=True, run_time=2))
+        self.wait()
+
+        self.play(
+            AnimationGroup(
+                FadeOut(fifty_one_squared, forty_nine_squared_1),
+                self.forty_nine_squared.animate.to_corner(UL),
+                lag_ratio=0.5
+            )
+        )
+        self.wait()
+
+    def get_formula(self):
+        '''
```

Comment:
Start on the same line
use double quote ticks
use imperative mode (like I do now), i.e. write instead of writes

Reply:
done